### PR TITLE
+ impl cni server and client

### PR DESCRIPTION
**Pull request by kebe7jun**

### CODE REVIEW

File: internal/cni-server/cni-plugin.go [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+	}
+	addrs := netnsEthsGetIPs(args.Netns)
+	if len(addrs) != 1 {
+		return fmt.Errorf("get ip address of %s error: res: %v", args.Netns, addrs)
+	}
+	lc := net.ListenConfig{
+		Control: func(network, address string, conn syscall.RawConn) error {
+			var operr error
+			if err := conn.Control(func(fd uintptr) {
+				if len(addrs) != 1 {
```
Comment (dddddai) [inline internal/cni-server/cni-plugin.go:159]: redundant check?

File: internal/cni-server/cni-plugin.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+	return err
+}
+
+func (s *server) CmdDelete(args *skel.CmdArgs) (err error) {
+	k8sArgs := plugin.K8sArgs{}
+	if err := types.LoadArgs(args.Args, &k8sArgs); err != nil {
+		return err
+	}
+	p := path.Join(s.bpfMountPath, "xdps", string(k8sArgs.K8S_POD_NAMESPACE), string(k8sArgs.K8S_POD_NAME))
+	os.RemoveAll(p)
```
[the remark this answers is not in the document]
Comment (kebe7jun) [reply internal/cni-server/cni-plugin.go:242]: good point.

[the remark this answers is not in the document]
Comment (dddddai) [reply internal/cni-server/cni-plugin.go:245]: Should we also clean up the directory(`/sys/fs/bpf/xdps/<pod-ns>/<pod-name>`)?

[the remark this answers is not in the document]
Comment (kebe7jun) [reply internal/cni-server/cni-plugin.go:245]: aha? `os.RemoveAll(p)` just do this, do I need to do anything else?

[the remark this answers is not in the document]
Comment (dddddai) [reply internal/cni-server/cni-plugin.go:245]: sorry, I misread it

**Review by dddddai — APPROVED** [review]

File: internal/cni-server/cni-plugin.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (dddddai) [reply internal/cni-server/cni-plugin.go:245]: Sorry, I misread it

**Pull request merged**